core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -194,7 +194,7 @@ abstract class RDD[T: ClassTag](
   @transient private var partitions_ : Array[Partition] = null
 
   /** An Option holding our checkpoint RDD, if we are checkpointed */
-  private def checkpointRDD: Option[RDD[T]] = checkpointData.flatMap(_.checkpointRDD)
+  private def checkpointRDD: Option[CheckpointRDD[T]] = checkpointData.flatMap(_.checkpointRDD)
 
   /**
```

Comment:
Since you are cleaning up the code, may i suggest another change. It is probably a good idea to mark RDD.checkpointData as volatile. Its a var, and assigning to it in `def checkpoint()` is not synchronized. Can cause race conditions.